src/utils/generator.js

```diff
@@ -111,11 +113,13 @@ type TemporalBuildNodeEntryArgs = {
   buildNode?: GeneratorBuildNodeFunction,
   dependencies?: Array<Generator>,
   isPure?: boolean,
+  isTerminal?: boolean,
 };
 
 class TemporalBuildNodeEntry extends GeneratorEntry {
   constructor(args: TemporalBuildNodeEntryArgs) {
     super();
+    this.isTerminal = !!args.isTerminal;
```

Comment:
Since isTerminal is declared to be optional, it is a bit surprising the see the !! here.

Reply:
It's optional in the `args`, but not optional in the `GeneratorEntry`.